src/test/rehydration.test.js

```diff
@@ -252,40 +247,20 @@ describe('rehydration', () => {
       const B = styled.div.withConfig({ componentId: 'TWO' })`color: red;`
       shallow(<B />)
       const styleTagsAfterRehydration = Array.from(document.querySelectorAll('style'))
-      expect(styleTagsAfterRehydration[0]).toBe(styleTags[0])
-      expect(styleTagsAfterRehydration[1]).toBe(styleTags[1])
+      expect(styleTagsAfterRehydration[0]).toEqual(styleTags[0])
 
       /* Only when new components are introduced (or a previous component
        * generates a new hash) does the style tag get replaced. */
       const C = styled.div.withConfig({ componentId: 'THREE' })`color: green;`
       shallow(<C />)
-      const styleTagsAfterAddition = Array.from(document.querySelectorAll('style'))
-
-      /* The first tag is unchanged */
-      expect(styleTagsAfterAddition[0]).toBe(styleTags[0])
-      /* The local tag has been replaced */
-      expect(styleTagsAfterAddition[1]).not.toBe(styleTags[1])
-      /* But it is identical, except for... */
-      expect(styleTagsAfterAddition[1].outerHTML).toEqual(
-        styleTags[1].outerHTML
-          /* ...the new data attribute for the new classname "c"... */
-          .replace(new RegExp(`${SC_ATTR}="a b"`), `${SC_ATTR}="a b c"`)
-          /* ...and the new CSS before the closing tag.  */
-          .replace(/(?=<\/style>)/, '\n/* sc-component-id: THREE */\n.THREE {}.c{color:green;}')
-      )
-
-      /* Note: any future additions don't replace the style tag */
-      const D = styled.div.withConfig({ componentId: 'TWO' })`color: tomato;`
-      shallow(<D />)
-
-      expect(Array.from(document.querySelectorAll('style'))[1]).toBe(styleTagsAfterAddition[1])
-
-      /* The point being, now we have a style tag we can inject new styles in the middle! */
+
+      /* the order stays correct and the styles are unharmed
+        * NOTE: during rehydration the empty rules are stripped out however */
```

Comment:
Why does rehydration strip the empty rules now? I'm pretty sure we still want those, at least in development.

Reply:
That’s because we’re using stylis to split the rules, so for instance “.A { }” isn’t sent over during SSR. It’s not really worth keeping it around if that implies that we need to meddle with what stylis does currently.

You can target those components regardless of whether the rule is present, since targeting the Button’s component ID for example depends on their className props, not on whether a rule for it is present